builtin/logical/database/mocks_test.go

```diff
@@ -36,8 +36,7 @@ func (m *mockNewDatabase) DeleteUser(ctx context.Context, req v5.DeleteUserReque
 }
 
 func (m *mockNewDatabase) Type() (string, error) {
-	args := m.Called()
-	return args.String(0), args.Error(1)
+	return "mock", nil
```

Comment:
Is there a reason why we're hardcoding the return here? 

I don't think we're currently doing this, but this wouldn't allow us to mock up errors from this method on tests if we needed to via something like `mockDB.On("Type").Return("", errors.New("error from type"))`.

Reply:
If we don't, we have to add `mockDB.On("Type")` to nearly every method or the tests will panic, since we use it in metrics collection.

Since this isn't used for much else that I saw, it seemed safer to just hardcode it here.